symphony-bdk-template/symphony-bdk-template-api/src/main/java/com/symphony/bdk/template/api/TemplateResolver.java

```diff
@@ -0,0 +1,116 @@
+package com.symphony.bdk.template.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * Class to ease resolution of a {@link Template} based on a {@link TemplateEngine} and a name,
+ * which can be a built-in template name, a path to a resource in the classpath,
+ * a path to a file on the file system or a URL to a template file.
+ */
+@Slf4j
+@API(status = API.Status.INTERNAL)
+public class TemplateResolver {
+  private final TemplateEngine templateEngine;
+
+  /**
+   * Constructor which uses the default {@link TemplateEngine} returned by {@link TemplateEngine#getDefaultImplementation()}
+   * in order to resolve templates.
+   */
+  public TemplateResolver() {
+    this(TemplateEngine.getDefaultImplementation());
+  }
+
+  /**
+   * Constructor which uses the {@link TemplateEngine} put in parameter.
+   * @param templateEngine the {@link TemplateEngine} instance to use to resolve and create {@link Template} instances.
+   */
+  public TemplateResolver(TemplateEngine templateEngine) {
+    this.templateEngine = templateEngine;
+  }
+
+  /**
+   * Looks for a template in this order : in the built-in templates, from a URL if the string has a valid URL format,
+   * in the classpath, in the filesystem
+   * @param template can be the name of a built-in template, a path to a template file in the classpath or in the file system
+   *                 or a URL to a template file
+   * @return a new {@link Template} instance corresponding to the template parameter
+   * @throws TemplateException if no template has been found.
+   */
+  public Template resolve(String template) throws TemplateException {
+    Template resolved = null;
+    if (isBuiltInTemplate(template)) {
+      resolved = tryFetchBuiltInTemplate(template);
+    }
+
+    if (resolved == null && isValidUrlSyntax(template)) {
```

Comment:
is there a reason to check the URL vs just trying like it is done for the others?

Reply:
I was thinking it is more efficient to check for the syntax instead of trying to connect to the URL. But indeed, if the URL is invalid, I guess it will fail fast.